src/rivian/ws_monitor.py

```diff
@@ -151,7 +152,10 @@ async def _receiver(self) -> None:
                         self._connection_ack.set()
                     elif data_type == "next":
                         if (_id := data.get("id")) in self._subscriptions:
-                            self._subscriptions[_id][0](data)
+                            if inspect.iscoroutinefunction(self._subscriptions[_id][0]):
```

Comment:
Please change the type of `_subscriptions` to allow for an Awaitable. You'll also have to adjust the typing anywhere that modifies `_subscriptions` to make sure it is appropriate as well.

Reply:
@natekspencer i don't think it is needed since async function is also Callable type same as normal function.

Reply:
Oh actually, yeah you're probably right. I need more coffee 😅

Reply:
It looks like mypy doesn't like it still though. But maybe you just need to do something like `_fn = self._subscriptions[_id][0]` and then the logic you added

Reply:
@natekspencer Fixed, it is passing mypy now